src/testing/jest/jest-adapter.ts

```diff
@@ -0,0 +1,74 @@
+import { JestPreprocessor, JestPuppeteerEnvironment, JestTestRunner } from './jest-apis';
+
+/**
+ * Base class for Jest-version specific code implementations that interact with Stencil.
+ *
+ * It is expected that there exists a Jest version-specific implementation for this adapter in each version-specific
+ * directory Stencil supports.
+ */
+export abstract class JestAdapter {
+  // TODO(STENCIL-961): Fix build validation when types are pulled in from `@stencil/core/declarations`
```

Comment:
Importing from `@stencil/core` or another subdirectory, the `test.dist` scripts begin to fail. I'm kicking the can down the road a little here on that, in case a better design reveals itself as I work on v28+ implementatons